lua/lualine/components/filetype.lua

```diff
@@ -8,7 +10,12 @@ FileType.update_status = function(self)
     local ok, devicons = pcall(require, 'nvim-web-devicons')
     if ok then
       local f_name, f_extension = vim.fn.expand('%:t'), vim.fn.expand('%:e')
-      self.options.icon = devicons.get_icon(f_name, f_extension)
+      icon, icon_highlight_group = devicons.get_icon(f_name, f_extension)
+      highlight_color = vim.fn.synIDattr(vim.fn.hlID(icon_highlight_group), 'fg')
+      self.options.icon_highlight = highlight.create_component_highlight_group(
```

Comment:
You can add a check with utils.utils.highlight_exists() that way you won't need to create new highlight on every load. maybe create hl_name like 'lualine_'..icon_highlight_group check if it exists if it doesn't create it . That way  all filetype colors won't have to go through same name .